build-system/tasks/performance/rewrite-script-tags.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const {
+  CONTROL,
+  EXPERIMENT,
+  copyToCache,
+  downloadToDisk,
+  urlToCachePath,
+} = require('./helpers');
+const {JSDOM} = require('jsdom');
+
+/**
+ * Lookup URL from cache and rewrite URLs to build from working branch
+ *
+ * @param {string} url
+ */
+async function useLocalScripts(url) {
+  const cachePath = urlToCachePath(url, EXPERIMENT);
+  const document = fs.readFileSync(cachePath);
+  const dom = new JSDOM(document);
+
+  const scripts = Array.from(dom.window.document.querySelectorAll('script'));
+  for (const script of scripts) {
+    if (script.src.startsWith('https://cdn.ampproject.org')) {
```

Comment:
Instead of hard-coding the cdn url, does it make sense to use `urls.cdn` or `urls.cdnProxyRegex` from here so that this code is immune to future changes in the proxy URL?

https://github.com/ampproject/amphtml/blob/7f870b9a4b2933d8a186541af5fc7d326a10a88c/src/config.js#L37-L46

Reply:
Can't find a way to reuse this const because `src` is using es6 modules and `build-system` is using node require. However moved to a const.